spatialos-sdk/src/worker/component.rs

```diff
@@ -237,14 +239,15 @@ pub struct ComponentDatabase {
 
 impl ComponentDatabase {
```

Comment:
I think the ComponentDatabase type is completely redundant at this point. Might be worth moving the code that generates a vtable array into a private free standing function that is called by the connection/snapshot `new` functions when generating a vtable to pass to the C API.

Reply:
Yeah I was having a similar thought - was looking into if we can just expose a single static `ComponentDatabase` instance that any type that requires it could reference, but `lazy_static` and `* c_void` are not compatible. 

Equally, the cost of alloc'ing a vtable each time its required isn't likely to be particularly high

Reply:
Yeah, a vtable is just a 16 function pointers for each component

Reply:
We could probably switch to having an optional flag on `WorkerConnection` like `use_vtable_serialization` that enables vtable registration, and then gather the vtables internally.